src/Sentry/Envelopes/ISerializable.cs

```diff
@@ -7,7 +7,7 @@ namespace Sentry.Protocol.Envelopes
     /// <summary>
     /// Represents a serializable entity.
     /// </summary>
-    internal interface ISerializable
+    public interface ISerializable
```

Comment:
I think this was another reason @Tyrrrz wanted to make things internal.
He had some plans to refactor this so types don't need to implement this in order to be serialized.

Reply:
That's most likely something that goes beyond of my understanding of the code vision of the library right now. I'm happy to help, but if there was an idea how to do this already, I might not think in the direction @Tyrrrz wanted to bring this, so I think it's better to wait for him.

Reply:
As a temp workaround to avoid exposing `ISerializable`, let's make:

- `EnvelopeItem.Payload` internal (or even private? can we do that?)
- `EnvelopeItem.ctor` internal (we won't expose a public constructor for envelope)

It's ugly but I think exposing `ISerializable` right now will constrain us too much. I don't like exposing `Envelope` either but there's no way around it if we want to expose `ITransport`.